index.js

```diff
@@ -216,57 +126,75 @@ export default class OneSignal {
         }
     }
 
-    /* deprecated */
-    static configure() {
-        console.warn("OneSignal: the `configure` method has been deprecated. The `ids` event is now triggered automatically.");
-    }
-
-    static init(appId, iOSSettings) {
-        if (!checkIfInitialized()) return;
+    static promptForPushNotificationPermissions(handler) {
+        if (!checkIfInitialized(RNOneSignal)) return;
 
         if (Platform.OS === 'ios') {
-            RNOneSignal.initWithAppId(appId, iOSSettings);
+            RNOneSignal.promptForPushNotificationPermissions(handler);
         } else {
-            RNOneSignal.init(appId);
+            console.log('This function is not supported on Android');
         }
     }
 
-    static checkPermissions(callback) {
-        if (!checkIfInitialized()) return;
+    static disablePush(disable) {
+        if (!checkIfInitialized(RNOneSignal)) return;
 
-        if (Platform.OS === 'ios') {
-            invariant(
-                typeof callback === 'function',
-                'Must provide a valid callback'
-            );
-            RNOneSignal.checkPermissions(callback);
-        } else {
-            console.log("This function is not supported on Android");
-        }
+        RNOneSignal.disablePush(disable);
+    }
+
+    static setLocationShared(shared) {
+        if (!checkIfInitialized(RNOneSignal)) return;
+
+        RNOneSignal.setLocationShared(shared);
     }
 
-    static promptForPushNotificationPermissions(callback) {
-        if (!checkIfInitialized()) return;
+    static promptLocation() {
+        if (!checkIfInitialized(RNOneSignal)) return;
+
+        //Supported in both iOS & Android
+        RNOneSignal.promptLocation();
+    }
+
+    /* D E V I C E  I N F O */
+
+    /**
+     * Gets the device state.
+     * // TO DO: add more details here
+     */
+    static getDeviceState() {
+        // TODO: check if we need to return promise?
+        if (!checkIfInitialized(RNOneSignal)) return Promise.resolve();
+        return RNOneSignal.getDeviceState();
+    }
+
+    static checkPermissions(handler) {
+        if (!checkIfInitialized(RNOneSignal)) return;
 
         if (Platform.OS === 'ios') {
-            RNOneSignal.promptForPushNotificationPermissions(callback);
+            isValidCallback(handler);
+            RNOneSignal.checkPermissions(handler);
         } else {
-            console.log('This function is not supported on Android');
+            console.log("This function is not supported on Android");
         }
     }
 
-    static getPermissionSubscriptionState(callback) {
-        if (!checkIfInitialized()) return;
+    static getPermissionSubscriptionState(handler) {
```

Comment:
Remove, as it was replaced by `getDeviceState`

Reply:
Done.